docs/neopixel.rst

```diff
@@ -0,0 +1,69 @@
+Neopixel
+****
+
+.. py:module:: neopixel
+
+The ``neopixel`` module lets you use Neopixel (WS2812) individually addressable RGB LED strips with the Microbit.    
+Note to use the ``neopixel`` module, you need to import it separately with::
+
+    import neopixel
+
+.. warning::
+    
+    Do not use the 3v connector on the Microbit to power any more than 8 Neopixels at a time.
```

Comment:
It depends how you power the micro:bit. If you power it from the USB, then
you are ultimately limited by the 'negotiated current at enumeration time'.
See below:

Davids-Air-2:~ davidw$ system_profiler SPUSBDataType > after

Davids-Air-2:~ davidw$ system_profiler SPUSBDataType > before

Davids-Air-2:~ davidw$ diff before after

20a21,38

>         MBED CMSIS-DAP:

>

>           Product ID: 0x0204

>           Vendor ID: 0x0d28

>           Version: 10.00

>           Serial Number: 9900023431864e45001600100000006300000000cc4d28bd

>           Speed: Up to 12 Mb/sec

>           Manufacturer: MBED

>           Location ID: 0x14100000 / 5

>           *Current Available (mA): 500*

*>           Current Required (mA): 500*

>           Capacity: 8.5 MB (8,454,144 bytes)

>           Removable Media: Yes

>           Detachable Drive: Yes

>           BSD Name: disk2

>           Partition Map Type: Unknown

>           S.M.A.R.T. status: Not Supported


You are then further limited by whatever the on-board power path is through
the freescale (as it regulates 5V down to 3V) - you could check the data
sheet for that part, as it's listed on some of the blogs what part number
this device is.


The story *might* be different when running from a battery - I doubt there
is even a regulator at that point, because otherwise you'd be regulating 3V
to, erm, 3V, which is pointless unless you ever expect a higher input
voltage. Legend has it though that a certain person ran it from 6V and got
away with it, which might suggest regulation is present, although I'm sure
that's out of spec. Check the freescale datasheet, you'll probably get the
best answer from that.


NeoPixels are designed to run from 5V, they *usually* work from 3V. So the
current characteristics might be different when below the rated voltage.


However, I would say that if the BBC say the maximum current from that pin
should be 100mA, then it is advisable to design circuits with that in mind,
and recommend an external power supply (another battery pack) for more than
100mA of NeoPixels.